src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const objectsInGroup: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyObject = String(item[key]);
+
+    if (keyObject in objectsInGroup) {
```

Comment:
Consider using `hasOwnProperty` instead of `in` for better performance and to avoid issues with inherited properties. 🚀